src/NuGet.Core/NuGet.Commands/RestoreCommand/Logging/PackageSpecificWarningProperties.cs

```diff
@@ -107,5 +153,37 @@ public bool Contains(NuGetLogCode code, string libraryId, NuGetFramework framewo
                 libraryIdsAndFrameworks.TryGetValue(libraryId, out var frameworkSet) &&
                 frameworkSet.Contains(framework);
         }
+
+        public override int GetHashCode()
+        {
+            var hashCode = new HashCodeCombiner();
+
+            // Add a constant hash for all objects
+            hashCode.AddObject(1);
```

Comment:
This is acceptable as per the requirements on HashCode - https://stackoverflow.com/questions/19710028/what-to-return-when-overriding-object-gethashcode-in-classes-with-no-immutable